packages/playwright-core/src/server/bidi/bidiPage.ts

```diff
@@ -193,18 +194,33 @@ export class BidiPage implements PageDelegate {
     this._page.frameManager.frameRequestedNavigation(frameId, params.navigation!);
   }
 
-  private _onNavigationCommitted(params: bidi.BrowsingContext.NavigationInfo) {
+  private async _onNavigationCommitted(params: bidi.BrowsingContext.NavigationInfo) {
     const frameId = params.context;
-    this._page.frameManager.frameCommittedNewDocumentNavigation(frameId, params.url, '', params.navigation!, /* initial */ false);
+    const namePromise = this._frameNamePromises.get(frameId) ?? '';
+    await this._page.frameManager.frameCommittedNewDocumentNavigation(frameId, params.url, namePromise, params.navigation!, /* initial */ false);
+    this._frameNamePromises.delete(frameId);
   }
 
-  private _onDomContentLoaded(params: bidi.BrowsingContext.NavigationInfo) {
+  private async _onDomContentLoaded(params: bidi.BrowsingContext.NavigationInfo) {
```

Comment:
Event listeners should be synchronous, we don't want to mess up the event order due to random awaits.

The frame name usually doesn't change, so we could query it on frame attach and if it arrives before the first navigation it will automatically propagate to the clients. This would be racy in some cases, but doesn't require async event handlers. We now have locators (and frame locators) that allow to select iframe elements and don't depend on this method, this could be a fallback for the folks that need to find a frame.